### DISCUSSION TITLE
Improve static js cache files

### DISCUSSION THREAD

**Discussion by Emiliano-Bucci** [DuCanhGH/next-pwa#102] [General]
Hi! I've noticed that if I implement the caching of the following assets -> `/\/_next\/static.+\.js$/i` js files are going to be cached on first sw registration, which could end up with lots of network requests on first load. Is there a way to dynamic cache this files, like for example when they're used by the app? 

I've lots of splitted js chunks that initially are not used by the app but still are cached. 

**Update**: I was thinking now that workers works in background thread without interfering with main thread and blocking UI. So I guess that all the assets requests are not a problem, that is correct? Even with insights reports?

Thanks!

**Comment by DuCanhGH** [comment]
@Emiliano-Bucci 

> Update: I was thinking now that workers works in background thread without interfering with main thread and blocking UI. So I guess that all the assets requests are not a problem, that is correct? Even with insights reports?

They shouldn't block the UI, but if the number of network requests is a problem for you, you can use `buildExcludes` to remove some or all JS files from the precache list. Note that pages will probably be broken when visited offline in that case though.
